hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientProxy.java

```diff
@@ -39,13 +39,17 @@
  */
 public abstract class ClientProxy implements DistributedObject {
 
+    protected final ClientContext context;
     protected final String name;
+
     private final String serviceName;
-    private volatile ClientContext context;
+    private final SerializationService serializationService;
```

Comment:
I'm also storing the `SerializationService` as field, so we can use it directly in the code.